contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -47,448 +43,251 @@ contract FraudProofFacet is StateChannelCommon {
+            blockDoubleSignProof.block2.encodedBlock,
+            blockDoubleSignProof.block2.signature
+        );
+        if(signer1 != signer2) {
+            return address(0);
+        }
+        return signer1; 
     }
 
-    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
-        // Implementation for handling BlockEmptyBlock proof
-        return address(0); // Replace with actual logic
+    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) pure internal returns (address) {
+        BlockEmptyProof memory blockEmptyProof = abi.decode(encodedProof, (BlockEmptyProof));
+        Block memory fraudBlock = abi.decode(blockEmptyProof.emptyBlock.encodedBlock, (Block));
+
+        if(fraudProofVerificationContext.channelId != fraudBlock.transaction.header.channelId) {
+            return address(0);
+        }
+        if(fraudBlock.transaction.header.transactionCnt != uint(0)) {
+            return address(0);
+        }
+        address signer = StateChannelUtilLibrary.retriveSignerAddress(
+            blockEmptyProof.emptyBlock.encodedBlock,
+            blockEmptyProof.emptyBlock.signature
+        );
+        return signer;
     }
 
     function _handleBlockInvalidStateTransition(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
-        // Implementation for handling BlockInvalidStateTransition proof
-        return address(0); // Replace with actual logic
+        BlockInvalidStateTransitionProof memory blockInvalidSTProof = abi.decode(encodedProof, (BlockInvalidStateTransitionProof));
+        Block memory fraudBlock = abi.decode(blockInvalidSTProof.invalidBlock.encodedBlock, (Block));
+        
+        if(fraudProofVerificationContext.channelId != fraudBlock.transaction.header.channelId) {
+            return address(0);
+        }
+
+        (bool isTransitionValid, bytes memory encodedModifiedState) = AStateChannelManagerProxy(address(this)).executeStateTransitionOnState(
+            fraudProofVerificationContext.channelId,
+            blockInvalidSTProof.encodedLatestState,
+            fraudBlock.transaction
+        );
+
+        if (isTransitionValid) {
+            return address(0);
+        } 
+
+        if(fraudBlock.stateSnapshotHash == keccak256(encodedModifiedState)){
+            return address(0);
+        }       
+        address signer = StateChannelUtilLibrary.retriveSignerAddress(
+            blockInvalidSTProof.invalidBlock.encodedBlock,
+            blockInvalidSTProof.invalidBlock.signature
+        );
+        return signer;
     }
 
     function _handleBlockOutOfGas(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
         // Implementation for handling BlockOutOfGas proof
```

Comment:
This could also be done in the same way as auditing with a low level function call that has a gas limit (the same limit on the client and on-chain) so if the gas limit is used the execution would fail and this would fall in the category of an 'invalidStateTransition'